source/common/http/BUILD

```diff
@@ -527,6 +527,7 @@ envoy_cc_library(
         "//source/common/common:utility_lib",
         "//source/common/protobuf:utility_lib",
         "//source/common/runtime:runtime_features_lib",
+        "@com_github_google_quiche//:quiche_common_structured_headers_lib",
```

Comment:
No change needed but calling out this adds quiche deps to the non-H3 build.  I think this makes sense for forwarding capsule protocol over H2 in non-H3 builds, just want to note it in case others disagree

Reply:
Right, I added this module to parse the structured field value (RFC 8941) of Capsule-Protocol header. The header field can be present regardless of the HTTP version (across H1/2/3). I can replace this dependency by implementing a simple parsing logic specifically for this header field if that is desirable. I will appreciate other reviewers' comments.  

Related to this, I should check this header field regardless of the HTTP version in the next PR to enable HTTP Datagram support. Where would be the best place to do this check? Should I create a http filter for this purpose?